pkg/scheduler/framework/plugins/README.md

```diff
@@ -0,0 +1,116 @@
+# Scheduler Framework Plugins
+
+## Creating a new in-tree plugin
+
+TODO: https://github.com/kubernetes/kubernetes/issues/95156
+
+## Making an in-tree plugin configurable through `KubeSchedulerConfiguration`
+
+You can give users the ability to configure parameters in scheduler plugins using
+[`KubeSchedulerConfiguration`](https://kubernetes.io/docs/reference/scheduling/config/).
+This section will cover how you can add arguments to existing in-tree plugins.
+Let's assume the plugin is called `FooPlugin` and we want to add an optional
+integer parameter named `BarParam`.
+
+First, we need to define a struct type named `FooPluginArgs`. In
+`pkg/scheduler/apis/config/types_pluginargs.go`, which defines the scheduler's
+internal struct type for passing config arguments:
+
+```go
+// Defined in pkg/scheduler/apis/config/types_pluginargs.go.
+type FooPluginArgs struct {
+	metav1.TypeMeta
```

Comment:
I think it can be beneficial to briefly explain why `TypeMeta` is needed.

Reply:
Done, but I'm not too familiar Kubernetes persisting mechanism, so PTAL to see if my comment is correct.

Reply:
I wouldn't put it as a comment in the code example, but as a separate paragraph. Maybe something like:

```
Please note that we embed `k8s.io/apimachinery/pkg/apis/meta/v1/TypeMeta` to include API metadata for versioning and persistence.
```